comp/otelcol/otlp/components/exporter/serializerexporter/consumer_collector.go

```diff
@@ -0,0 +1,118 @@
+)
+
+type collectorConsumer struct {
+	*serializerConsumer
+	seenHosts    map[string]struct{}
+	seenTags     map[string]struct{}
+	buildInfo    component.BuildInfo
+	gatewayUsage *attributes.GatewayUsage
+	// getPushTime returns a Unix time in nanoseconds, representing the time pushing metrics.
+	// It will be overwritten in tests.
+	getPushTime func() uint64
+}
+
+var _ SerializeConsumer = (*collectorConsumer)(nil)
+
+func (c *collectorConsumer) addRuntimeTelemetryMetric(_ string, languageTags []string) {
+	timestamp := c.getPushTime()
+	buildTags := tagsFromBuildInfo(c.buildInfo)
+	series := c.series
+	for host := range c.seenHosts {
+		// Report the host as running
+		runningMetric := exporterDefaultMetrics("metrics", host, timestamp, buildTags)
+		if c.gatewayUsage != nil {
+			series = append(series, gatewayUsageGauge(timestamp, host, buildTags, c.gatewayUsage))
+		}
+		series = append(series, runningMetric)
+	}
+
+	var tags []string
+	tags = append(tags, buildTags...)
+	for tag := range c.seenTags {
+		tags = append(tags, tag)
+	}
+	runningMetrics := exporterDefaultMetrics("metrics", "", timestamp, tags)
+	series = append(series, runningMetrics)
+
+	for _, lang := range languageTags {
+		tags := append(buildTags, "language:"+lang) //nolint:gocritic
+		runningMetric := exporterDefaultMetrics("runtime_metrics", "", timestamp, tags)
+		series = append(series, runningMetric)
+	}
+	c.series = series
+}
+
+func (c *collectorConsumer) addTelemetryMetric(_ string) {
+}
+
+// ConsumeHost implements the metrics.HostConsumer interface.
+func (c *collectorConsumer) ConsumeHost(host string) {
+	c.seenHosts[host] = struct{}{}
+}
+
+// ConsumeTag implements the metrics.TagsConsumer interface.
+func (c *collectorConsumer) ConsumeTag(tag string) {
+	c.seenTags[tag] = struct{}{}
+}
+
+// exporterDefaultMetrics creates built-in metrics to report that an exporter is running
+func exporterDefaultMetrics(exporterType string, hostname string, timestamp uint64, tags []string) *metrics.Serie {
+	metrics := &metrics.Serie{
+		Name: fmt.Sprintf("otel.datadog_exporter.%s.running", exporterType),
+		Points: []metrics.Point{
+			{
+				Ts:    float64(timestamp),
+				Value: 1.0,
+			},
+		},
+		Host:   hostname,
+		MType:  metrics.APIGaugeType,
+		Tags:   tagset.CompositeTagsFromSlice(tags),
+		Source: metrics.MetricSourceOpenTelemetryCollectorUnknown,
+	}
+	return metrics
+}
+
+// gatewayUsageGauge creates a gauge metric to report if there is a gateway
+func gatewayUsageGauge(timestamp uint64, hostname string, tags []string, gatewayUsage *attributes.GatewayUsage) *metrics.Serie {
+	metrics := &metrics.Serie{
+		Name: "datadog.otel.gateway",
+		Points: []metrics.Point{
+			{
+				Ts:    float64(timestamp),
+				Value: gatewayUsage.Gauge(),
+			},
+		},
+		Host:   hostname,
+		MType:  metrics.APIGaugeType,
+		Tags:   tagset.CompositeTagsFromSlice(tags),
+		Source: metrics.MetricSourceOpenTelemetryCollectorUnknown,
+	}
+	return metrics
+}
```

Comment:
I think we want this in embedded collector too cc @ogaca-dd

Reply:
may be it should be a different metric name in embedded collector.